### ISSUE TITLE
feat: model selection (XGBoost vs LightGBM vs RandomForest)

### ISSUE THREAD

**Issue by omar-tarsbot** [otter-ml/otter#14]
## What
Implement `otter/train/selector.py` — tries multiple models and picks the best.

## Models to try
1. XGBoost (primary — usually wins)
2. LightGBM (fast, good on large datasets)
3. RandomForest (baseline, interpretable)
4. LogisticRegression (baseline for binary classification)

## Selection logic
- Quick 3-fold CV on 20% of data to eliminate losers
- Full Optuna tuning on top-2 models
- Winner = highest AUC on held-out test set

## Output
```python
@dataclass
class ModelSelectionResult:
    winner: str           # 'XGBoost'
    winner_auc: float
    all_results: dict     # {model_name: auc}
    winner_model: any     # fitted model object
```

## UX (shown during training)
```
✓ Model selection  XGBoost 0.91 | LightGBM 0.89 | RandomForest 0.84
```

## Part of v0.1 — depends on #13 (features)
